app/controllers/provider/admin/backend_apis/metrics_controller.rb

```diff
@@ -17,19 +11,41 @@ def index
   end
 
   def new
-    render template: '/api/metrics/new'
+    @metric = collection.build
   end
 
   def create
-    # TODO: https://issues.jboss.org/browse/THREESCALE-3089
+    @metric = collection.build(metric_params.merge(owner: @backend_api))
```

Comment:
Why do we need owner here?
Is it because of       `find_backend_api_metric_by(metric_id).children`?
Meaning that children (methods) of a metric should have the same owner right?

If we can fix it somehow in `acts_as_tree` it would probably be better

Reply:
Right. I was lazy. It is because of `find_backend_api_metric_by `. We need to fix it there.